slf4j-mdc-propagation/pom.xml

```diff
@@ -27,7 +27,7 @@
 
     <!-- Artifact identification -->
     <artifactId>mdc-propagation</artifactId>
-    <name>Propagation of MDC</name>
+    <name>Propagation of SLF4J MDC</name>
```

Comment:
Please also rename artifactId to `slf4j-mdc-propagation` to maintain consistency.

We can add a `mdc-propagation` module to the relocations that redirects users to the new artifact.

Reply:
Should we rename it then to `slf4j-propagation` in case this library covers aspects other than MDC in the future as well (this was also the reason why I chose `log4j2-propagation`)?

Reply:
I'd prefer it if the rename were tackled in a separate PR, so there are less different topics to discuss in this one. (I created #209 for the Slf4j module rename)